### PR TITLE
Gpt output loss

### PR DESCRIPTION
**Pull request by Yam0214**
### PR types

New features

### PR changes

Models

### Description

Add more outputs and loss to GPT model.

### CODE REVIEW

File: paddlenlp/transformers/gpt/modeling.py [hunk 1]
```diff
@@ -1119,31 +1194,65 @@ def forward(self,
...
-        logits = self.lm_head(encoder_outputs)
-
-        if use_cache:
-            return logits, cached_kvs
-        else:
-            return logits
+                           cache=cache,
+                           output_attentions=output_attentions,
+                           output_hidden_states=output_hidden_states,
+                           return_dict=True)
```
Comment (FrostML) [inline paddlenlp/transformers/gpt/modeling.py]: `return_dict=True` 这里是不是忘记了

File: paddlenlp/transformers/gpt/modeling.py [hunk 2]
```diff
@@ -1276,13 +1408,37 @@ def forward(self, input_ids, position_ids=None, attention_mask=None):
...
-        sequence_output = self.dropout(sequence_output)
-        logits = self.classifier(sequence_output)
-        return logits
+        sequence_output = self.gpt(
+            input_ids,
+            position_ids=position_ids,
+            attention_mask=attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=True,
```
Comment (FrostML) [inline paddlenlp/transformers/gpt/modeling.py]: 同上

File: paddlenlp/transformers/gpt/modeling.py [hunk 3]
```diff
@@ -1340,20 +1526,50 @@ def forward(self, input_ids, position_ids=None, attention_mask=None):
         # sequence_output shape [bs, seq_len, hidden_size]
         sequence_output = self.gpt(input_ids,
                                    position_ids=position_ids,
-                                   attention_mask=attention_mask)
+                                   attention_mask=attention_mask,
+                                   use_cache=use_cache,
+                                   output_attentions=output_attentions,
+                                   output_hidden_states=output_hidden_states,
+                                   return_dict=True)
```
Comment (FrostML) [inline paddlenlp/transformers/gpt/modeling.py]: 同上

Comment (Yam0214) [reply paddlenlp/transformers/gpt/modeling.py]: 三个地方都已修改

**Review by FrostML — DISMISSED** [review]

**Review by FrostML — DISMISSED** [review]

File: paddlenlp/transformers/codegen/modeling.py [hunk 4]
```diff
@@ -417,57 +420,61 @@ def forward(
...
+                    An instance of :class:`~paddlenlp.transformers.model_outputs.BaseModelOutputWithPastAndCrossAttentions` if
+                    `return_dict=True`. Otherwise it returns a tuple of tensors corresponding
+                    to ordered and not None (depending on the input arguments) fields of
+                    :class:`~paddlenlp.transformers.model_outputs.BaseModelOutputWithPastAndCrossAttentions`.
+                    Especially, When `return_dict=output_hidden_states=output_attentions=False` and `cache=None`,
+        <<<<<<< HEAD
+                    returns a tensor representing the output of :class:`CodeGenModel`.
+        =======
+                    returns a tensor representing the output of :class:`UnifiedTransformerModel`.
+        >>>>>>> 56e1ad9e7bfc89651b7c317a6aa1b68fb42d15da
```
Comment (FrostML) [inline paddlenlp/transformers/codegen/modeling.py]: 冲突处理下

File: paddlenlp/transformers/model_outputs.py [hunk 5]
```diff
@@ -14,15 +14,15 @@
...
 import paddle
 from paddle import Tensor
-import numpy as np
-from typing import Optional, Tuple
-from collections import OrderedDict
-from dataclasses import fields, dataclass
-from typing import Any, Tuple, Optional
-from paddle.nn.layer.transformer import _convert_attention_mask, MultiHeadAttention
 from paddle.distributed.fleet.utils import recompute
+from paddle.nn.layer.transformer import MultiHeadAttention, _convert_attention_mask
```
Comment (FrostML) [inline paddlenlp/transformers/model_outputs.py]: 这里分开来写吧，`MultiHeadAttention` 官方文档里面不是这么使用的，而 `_convert_attention_mask ` 其实不算是个公开的官方 API
``` python
from paddle.nn import MultiHeadAttention
from paddle.nn.layer.transformer import _convert_attention_mask
```

**Review by FrostML — APPROVED** [review]
